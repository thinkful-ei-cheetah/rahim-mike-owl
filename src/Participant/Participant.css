li {
  display: flex;
}

.name {
  flex: 1;
}

.avatar {
  flex: 3;
}
