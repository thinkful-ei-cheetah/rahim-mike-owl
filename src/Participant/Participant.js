import React from 'react';
import './Participant.css';

export default function Participant(props) {
  return (
    <li>
      <div className='avatar'>
        <img src={props.avatar} alt={'avatar'} />
      </div>
      <div className='name'>
        <p>{props.name}</p>
        <p>{props.inSession}</p>
      </div>
    </li>
  );
}
