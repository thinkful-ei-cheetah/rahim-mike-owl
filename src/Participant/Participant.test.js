import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Participant from './Participant';

describe('Participant component', () => {
  let data;
  beforeEach(() => {
    data = {
      id: 1,
      name: 'Koren Templeton',
      avatar:
        'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1',
      inSession: true,
      onStage: true
    };
    return data;
  });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Participant />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders a participant', () => {
    const tree = renderer.create(
      <Participant
        name={data.name}
        avatar={data.avatar}
        inSession={data.inSession}
      />
    ).toJSON;
    expect(tree).toMatchSnapshot();
  });
});
