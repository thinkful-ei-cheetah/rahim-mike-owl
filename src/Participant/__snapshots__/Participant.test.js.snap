// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Participant component renders a participant 1`] = `[Function]`;
